dandischema/models.py

```diff
@@ -1133,42 +952,57 @@ class BareAssetMeta(CommonModel):
         "nskey": "dandi",
     }
 
+    @validator("digest")
+    def digest_etag(cls, values):
+        print(values)
+        try:
+            if len(values[DigestType.dandi_etag]) != 32:
+                raise ValueError
+        except (KeyError, ValueError):
+            raise ValueError("Digest must have an appropriate dandi-etag value.")
+        return values
+
 
-class AssetMeta(BareAssetMeta, Identifiable):
+class Asset(BareAsset):
     """Metadata used to describe an asset on the server."""
 
-    id: Optional[str] = Field(readOnly=True, description="Uniform resource identifier")
-    identifier: Optional[UUID4] = Field(readOnly=True, nskey="schema")
-    # on publish or set by server
-    contentUrl: Optional[List[HttpUrl]] = Field(None, readOnly=True, nskey="schema")
+    # all of the following are set by server
+    id: str = Field(readOnly=True, description="Uniform resource identifier")
+    identifier: UUID4 = Field(readOnly=True, nskey="schema")
+    contentUrl: List[HttpUrl] = Field(None, readOnly=True, nskey="schema")
 
 
 class Publishable(DandiBaseModel):
-    id: str = Field(readOnly=True, description="Uniform resource identifier")
     publishedBy: Union[HttpUrl, PublishActivity] = Field(
         description="The URL should contain the provenance of the publishing process.",
         readOnly=True,
         nskey="dandi",
     )  # TODO: formalize "publish" activity to at least the Actor
     datePublished: datetime = Field(readOnly=True, nskey="schema")
-    url: HttpUrl = Field(
-        readOnly=True, description="permalink to the item", nskey="schema"
-    )
+    schemaKey: Literal["Publishable"] = Field("Publishable", readOnly=True)
 
 
-class PublishedDandisetMeta(DandisetMeta, Publishable):
-    version: str = Field(readOnly=True, nskey="schema")
+class PublishedDandiset(Dandiset, Publishable):
     doi: str = Field(
         title="DOI",
         readOnly=True,
         regex=r"^10\.[A-Za-z0-9.\/-]+",
         nskey="dandi",
     )
-    assetsSummary: AssetsSummary = Field(readOnly=True, nskey="dandi")
+    url: HttpUrl = Field(
+        readOnly=True, description="permalink to the item", nskey="schema"
+    )
 
 
-class PublishedAssetMeta(AssetMeta, Publishable):
-    pass
+class PublishedAsset(Asset, Publishable):
+    @validator("digest")
+    def digest_bothhashes(cls, values):
+        try:
+            if len(values[DigestType.dandi_etag] + values[DigestType.sha2_256]) != 96:
+                raise ValueError
+        except (KeyError, ValueError):
+            raise ValueError("Digest must have both dandi-etag and sha2-256.")
```

Comment:
```suggestion
            raise ValueError("Digest must have both valid dandi-etag and sha2-256.")
```
I would have loved to stick actual values but decided not to bother yet - would need assignment to temp vars above